From 31ea9c63159cf2d9fb338a4fe9b725ed40c5463c Mon Sep 17 00:00:00 2001
From: David Wagner <david.wagner@intel.com>
Date: Mon, 6 May 2013 14:56:36 +0200
Subject: [PDK MR2 Temp] Do not rely on the BT state to elect the BT route

BZ: 117199

Instead, rely solely on the policy information regarding the device selection.

We used to use the 'bluetooth_enabled' parameter but this information was lost
in R4 and reintroduced in
frameworks/base/services/java/com/android/server/BluetoothManagerService.java
by http://android.intel.com:8080/#/c/91943/
However, this file is not delivered as source in the PDK.

This patch may be merged to mainline but this needs testing.

Change-Id: I714d5a36ce7c6b4e4fb930849c6899f04df7f10b
Signed-off-by: David Wagner <david.wagner@intel.com>
---
 .../AudioPlatformHardware_redhookbay.cpp           |    4 ++--
 .../AudioPlatformHardware_victoriabay.cpp          |    6 ------
 2 files changed, 2 insertions(+), 8 deletions(-)

diff --git a/audio_hw_configurable/audio_route_manager/AudioPlatformHardware_redhookbay.cpp b/audio_hw_configurable/audio_route_manager/AudioPlatformHardware_redhookbay.cpp
index 4e6a4a5..aa0ad04 100755
--- a/audio_hw_configurable/audio_route_manager/AudioPlatformHardware_redhookbay.cpp
+++ b/audio_hw_configurable/audio_route_manager/AudioPlatformHardware_redhookbay.cpp
@@ -717,7 +717,7 @@ public:
         // BT module must be off and as the BT is on the shared I2S bus
         // the modem must be alive as well to use this route
         // and it should be the one and only one available device as it does not have the priority
-        if (!_pPlatformState->isSharedI2SBusAvailable() || !_pPlatformState->isBtEnabled() ||
+        if (!_pPlatformState->isSharedI2SBusAvailable() ||
                 ((uidevices != 0) && (AudioSystem::popCount(uidevices) != 1))) {
 
             return false;
@@ -857,7 +857,7 @@ public:
 
         // BT module must be off and as the BT is on the shared I2S bus
         // the modem must be alive as well to use this route
-        if (!_pPlatformState->isModemAudioAvailable() || !_pPlatformState->isBtEnabled()) {
+        if (!_pPlatformState->isModemAudioAvailable()) {
 
             return false;
         }
diff --git a/audio_hw_configurable/audio_route_manager/AudioPlatformHardware_victoriabay.cpp b/audio_hw_configurable/audio_route_manager/AudioPlatformHardware_victoriabay.cpp
index 799cf49..b8f3fc8 100755
--- a/audio_hw_configurable/audio_route_manager/AudioPlatformHardware_victoriabay.cpp
+++ b/audio_hw_configurable/audio_route_manager/AudioPlatformHardware_victoriabay.cpp
@@ -672,12 +672,6 @@ public:
         _pPlatformState->setBtEnabled(true);
 #endif
 
-        // BT module must be on and as the BT is on the shared I2S bus
-        // the share bus must be available
-        if (!_pPlatformState->isBtEnabled()) {
-
-            return false;
-        }
         if (!bIsOut && (iMode == AudioSystem::MODE_IN_CALL)) {
 
             // In Voice CALL, the audio policy does not give any input device
-- 
1.7.4.1

